test/system/src/test/java/io/pravega/test/system/SegmentReaderAPITest.java

```diff
@@ -0,0 +1,468 @@
+        controllerURI = fetchControllerURI();
+        streamManager = StreamManager.create(Utils.buildClientConfig(controllerURI));
+
+        connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+        controller = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(Utils.buildClientConfig(controllerURI))
+                .build(), connectionFactory.getInternalExecutor());
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        controller.close();
+        connectionFactory.close();
+    }
+
+    @Test(timeout = 60000)
+    public void getNextStreamCutWithScaleUpTest() throws SegmentTruncatedException, ExecutionException, InterruptedException, TimeoutException {
+        String streamName = "testStreamSegment";
+        String streamScope = "testScopeSegment";
+        String readerGroupName = "testReaderGroupSegment";
+        String readerName = UUID.randomUUID().toString();
+
+        StreamConfiguration config = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build();
+        ClientConfig clientConfig = ClientConfig.builder()
+                .controllerURI(fetchControllerURI())
+                .build();
+        URI controllerURI = clientConfig.getControllerURI();
+
+        Stream stream = Stream.of(streamScope, streamName);
+
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(streamScope, controller, connectionFactory);
+
+        assertTrue(controller.createScope(streamScope).join());
+        assertTrue(controller.createStream(streamScope, streamName, config).join());
+
+        @Cleanup
+        BatchClientFactory batchClient = BatchClientFactory.withScope(streamScope, clientConfig);
+
+        List<SegmentRange> ranges = Lists.newArrayList(batchClient.getSegments(stream, StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+        List<Segment> list = ranges.stream().map(SegmentRange::getSegment).collect(Collectors.toList());
+        assertEquals(1, list.size());
+        log.info("Segment name ::{}", list.get(0).getScopedName());
+
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new UTF8StringSerializer(),
+                EventWriterConfig.builder().build());
+
+        // write events to stream 30 *10  = 300 bytes
+        write30ByteEvents(10, writer);
+        writer.flush();
+
+        Map<Segment, Long> positions = new HashMap<>();
+        positions.put(list.get(0), 0L);
+        StreamCut streamCut0 = new StreamCutImpl(Stream.of(streamScope, streamName), positions);
+        log.info("Initial stream streamCut0 {}", streamCut0);
+
+        //Requested next stream cut at a distance of 170 bytes, and getting 180 offset as a response.
+        StreamCut streamCut1 = batchClient.getNextStreamCut(streamCut0, 170L);
+        log.info("Next stream cut1 {}", streamCut1);
```

Comment:
Is there any assertion to be done after getting this streamcut? What do you expect at this point?